### PR TITLE
Clean up site, remove partners.

### PR DESCRIPTION
**Pull request by tackaberry**
A few things to clean up for revival 2024.  

I removed partners section. 

I removed Slack link for now until we can rebuild an invite process.  

I updated the  up procedure and added a better readme to include how to test locally.

As part of this effort I also renamed the master branch to main and added protection to the main branch to require a PR to merge and at least one approver.  Also I removed owner access from everyone except known contributors.

### CODE REVIEW

File: .gitignore
Comment (lchski) [inline .gitignore:1]: It's still checked in to the repo—I think it’s normal to check it in, though, so we could probably remove it from the `.gitignore`. Not a huge deal either way.

**Review by lchski — APPROVED** [review]
Looks good to me. Probably don’t need the `Gemfile.lock` added to the `.gitignore`, but it won’t have any negative effects.

:shipit: I’d say!

(with the caveat that my brew / ruby envs were taking a while to bring back up to date, so I haven’t actually tried building it locally, but I’m not too worried about that)

**Comment by tackaberry** [comment]
Ok gemfile.lock removed from gitignore.  Also not shown in this PR is fussing with local to get it running.

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ _site
 .sass-cache
 .jekyll-metadata
 .DS_Store
+Gemfile.lock
```
Comment (nschonni) [inline .gitignore:5]:
```suggestion

```

File: .ruby-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.1.2
```
Comment (nschonni) [inline .ruby-version:1]:
```suggestion
3.3.4
```

File: Gemfile [hunk 3]
```diff
@@ -8,20 +7,31 @@ source "https://rubygems.org"
...
-
+gem "minima", "~> 2.5"
 # If you want to use GitHub Pages, remove the "gem "jekyll"" above and
 # uncomment the line below. To upgrade, run `bundle update github-pages`.
 # gem "github-pages", group: :jekyll_plugins
-
 # If you have any plugins, put them here!
 group :jekyll_plugins do
-  gem "jekyll-feed", "~> 0.6"
+  gem "jekyll-feed", "~> 0.12"
```
Comment (nschonni) [inline Gemfile:20]: This is inlcuded with the `github-pages` gem, so isn't needed anymore

File: Gemfile [hunk 4]
```diff
@@ -8,20 +7,31 @@ source "https://rubygems.org"
...
 # This will help ensure the proper Jekyll version is running.
 # Happy Jekylling!
-gem "jekyll", "~> 3.6.0"
+# gem "jekyll", "~> 4.3.3"
+gem "github-pages", "~> 231", group: :jekyll_plugins
 
 # This is the default theme for new Jekyll sites. You may change this to anything you like.
-gem "minima", "~> 2.0"
-
+gem "minima", "~> 2.5"
```
Comment (nschonni) [inline Gemfile:14]: Included with `github-pages`

**Review by nschonni — COMMENTED** [review]
I think moving to `github-pages` makes sense. Most of the Gemfile could probably be wiped out since most of the things previously included are bundled, and the other things seem like working around local windows/JRuby issues

File: Gemfile (hunk 3 above)
Comment (tackaberry) [reply Gemfile:20]: Ok.

File: Gemfile (hunk 4 above)
Comment (tackaberry) [reply Gemfile:14]: Ok.

**Comment by tackaberry** [comment]
@nschonni @lchski Comments addressed.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,34 @@
...
+These instructions are adapted from "[Creating a GitHub Pages site with Jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll) and [Testing your GitHub Pages site locally with Jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll)."
+1. Install Ruby. For more information, see "[Installing Ruby](https://www.ruby-lang.org/en/documentation/installation/)" in the Ruby documentation.
+2. (Optional) Install [rbenv](https://github.com/rbenv/rbenv). Run `rbenv local 3.1.2` to set up a localized environment. 
+2. Install Bundler. For more information, see "[Bundler](https://bundler.io/)."
+
+```bash
+
+git clone https://github.com/YOWCT/yowct.github.io
+cd yowct.github.io
+rbenv local 3.1.2
```
Comment (nschonni) [inline README.md:25]: I think that will override the version in the `.ruby-version`. I think I use `rvm use .` locally, but I'm not sure the `rbenv` equivalent

Comment (tackaberry) [reply README.md]: I cleaned up the versioning for consistency. But kept in the code - could be helpful.

**Review by nschonni — APPROVED** [review]

**Review by lchski — APPROVED** [review]

**Pull request merged**